src/extension/SettingsPanel.ts

```diff
@@ -60,33 +64,75 @@ export class SettingsPanel {
       null,
       [],
     )
-
     this._panel.onDidDispose(this.dispose, null, this._disposables)
   }
 
+  /**
+   * opens a new settings panel in a new tab
+   * @param extensionUri uri of the extension folder
+   * @param editConfFile conf file to edit
+   */
+  private static _openNewPanel(
+    extensionUri: vscode.Uri,
+    editConfFile?: Record<string, unknown>,
+  ) {
+    const editFileName = editConfFile?.verify + ''
+    let confFileName = ''
+    if (editConfFile) {
+      confFileName += editFileName.replace(':', '.').replace('spec', 'conf')
+    }
+    const panel = vscode.window.createWebviewPanel(
+      'certoraSettings',
+      'Certora IDE Settings: ' + confFileName,
+      vscode.ViewColumn.One,
+      {
+        retainContextWhenHidden: true,
+        enableScripts: true,
+        localResourceRoots: [extensionUri],
+      },
+    )
+    SettingsPanel.currentPanel = new SettingsPanel(
+      panel,
+      extensionUri,
+      editConfFile,
+    )
+    this.allPanels.push(SettingsPanel.currentPanel)
+  }
+
+  /**
+   * renders a settings panel:
+   * if the settings panel is not an edit: create a new panel in a new tab
+   * else if the settings panel for that edit was not created, or created and deleted: create new
+   * else: open the already created panel tab
```

Comment:
It's great that you documented it. Let's make it a bit shorter:
Open a new webview panel when creating or editing conf files. If the panel is already opened (for editing an existing configuration file), show the same panel.